ams/ams-api/src/main/thrift/arctic_optimize_manager.thrift

```diff
@@ -119,7 +119,7 @@ service OptimizeManager {
 
     void ping()
 
-    OptimizeTask pollTask(1:i32 queueId, 2:JobId jobId, 3:string attemptId, 4:i64 waitTime)
+    OptimizeTask pollTask(1:i32 queueId, 2:JobId jobId, 3:string attemptId, 4:i64 waitTime, 5:i32 subtaskId)
```

Comment:
@zhoujinsong What do you think about the compatibility problems?

Reply:
> We have to check if there will be compatibility problems after we change this thrift interface method signature, in both two cases:
> 
> * optimizer with old interface -> AMS with new interface
> 
> Maybe we have to add a new method with `subTaskId`, and keep the old one for compatibility.

Compatible with this scenario should be relatively easy.


> * optimizer with new interface -> AMS with old interface.

This scenario is not easy to do, do you have any ideas?
@wangtaohz

Reply:
You can verify it by deploying different versions of Arctic and Optimizer, but since that the new interface has been added to address the compatibility issues, there is no need to do that.

Reply:
I have added the new optimizer to be compatible with the old ams implementation, as you say it not need to do that?
if yes, I will remove this part of the implementation.

Reply:
I mean there is no need to verify the compatibility issue, your implementation is applicable.